server/src/main/java/org/apache/druid/server/security/AuthorizationUtils.java

```diff
@@ -263,10 +263,6 @@ public static Access authorizeAllResourceActions(
       throw new ISE("No authorizer found with name: [%s].", authenticationResult.getAuthorizerName());
     }
 
-    if (authorizer instanceof AllowAllAuthorizer) {
-      return resources;
-    }
-
```

Comment:
Could we add a test to prevent someone accidentally making this addition in the future. I read the previous patch and thought this was a good optimization too.

Reply:
Good call, @suneet-s . I added a unit test in the new commit and in the process  I found that we can still skip authorizing the resources when AllowAllAuthorizer is configured.